test/Microsoft.AspNetCore.Mvc.Razor.Extensions.Test/ModelDirectiveTest.cs

```diff
@@ -236,26 +236,26 @@ private NamespaceDeclarationIntermediateNode FindNamespaceNode(IntermediateNode
 
         private RazorEngine CreateEngine()
         {
-            return RazorEngine.Create(b =>
-            {
-                // Notice we're not registering the ModelDirective.Pass here so we can run it on demand.
-                b.AddDirective(ModelDirective.Directive);
-
-                // There's some special interaction with the inherits directive
-                InheritsDirective.Register(b);
-            });
+            return CreateEngineCore();
         }
 
         private RazorEngine CreateDesignTimeEngine()
         {
-            return RazorEngine.CreateDesignTime(b =>
+            return CreateEngineCore(designTime: true);
+        }
+
+        private RazorEngine CreateEngineCore(bool designTime = false)
```

Comment:
This doesn't seem right. We don't have a concept like "design time engine" anymore

Reply:
Yea, this flag is not the input for the engine itself. Just using it to set the right value for the options.

Reply:
Then that's the right fix. They should not have the right value for options.

Reply:
Filed https://github.com/aspnet/Razor/issues/2275 to address this.